Core/include/Acts/EventData/SpacePointProxy.hpp

```diff
@@ -0,0 +1,60 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Utilities/HashedString.hpp"
+#include "Acts/Utilities/Holders.hpp"
+
+#include <any>
+#include <iostream>
+#include <string_view>
+#include <type_traits>
+
+namespace Acts {
+
+template <typename container_t>
+class SpacePointProxy {
+ public:
+  using ContainerType = container_t;
+  using ValueType = typename ContainerType::ValueType;
+
+ public:
+  // Never take the ownership of the container
+  SpacePointProxy(container_t&& container, std::size_t index) = delete;
+  // Only get the reference
```

Comment:
I don't think it will. The following code works:
```cpp
#include <iostream>

template<typename T>
class A {
public:
  A(const T&) {}
};

class B {};

int main() {
  B b;
  A<B> a(std::move(b));
}
```